### PR TITLE
integration with docker

### PR DESCRIPTION
**Pull request by aagarwal2286**
this pull request is to make this awesome repository docker compatible so that users may start working without running into environment setup issues.

### CODE REVIEW

**Comment by krzysiekpiasecki** [comment]
@aagarwal2286  There is an ERROR when `docker-compose up --build` after cloning your branch.

> ui    | npm ERR! code ERR_TLS_CERT_ALTNAME_INVALID
> ui    | npm ERR! errno ERR_TLS_CERT_ALTNAME_INVALID
> ui    | npm ERR! request to https://registry.npmjs.org/babel-polyfill failed, reason: Hostname/IP does not match certificate's altnames: Host: registry.npmjs.org. is not in the cert's altnames: DNS:a.sni.fastly.net, DNS:a.sni.global-ssl.fastly.net
> ui    | 
> ui    | npm ERR! A complete log of this run can be found in:
> ui    | npm ERR!     /root/.npm/_logs/2018-09-01T09_51_37_221Z-debug.log
> ui    | 
> ui    | > adminator@1.0.0 docker-dev /home/node/app
> ui    | > webpack-dev-server
> ui    | 
> ui    | sh: 1: webpack-dev-server: not found
> ui    | npm ERR! file sh
> ui    | npm ERR! code ELIFECYCLE
> ui    | npm ERR! errno ENOENT
> ui    | npm ERR! syscall spawn
> ui    | npm ERR! adminator@1.0.0 docker-dev: `webpack-dev-server`
> ui    | npm ERR! spawn ENOENT
> ui    | npm ERR! 
> ui    | npm ERR! Failed at the adminator@1.0.0 docker-dev script.
> ui    | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> ui    | npm WARN Local package.json exists, but node_modules missing, did you mean to install?
> ui    | 
> ui    | npm ERR! A complete log of this run can be found in:
> ui    | npm ERR!     /root/.npm/_logs/2018-09-01T09_51_37_536Z-debug.log


You should also change the name of the container.

**Review by krzysiekpiasecki — CHANGES_REQUESTED** [review]
Container name is to general and error occurs when running this container.

**Comment by aagarwal2286** [comment]
> ui | npm ERR! request to https://registry.npmjs.org/babel-polyfill failed, reason: Hostname/IP does not match certificate's altnames: Host: registry.npmjs.org. is not in the cert's altnames: DNS:a.sni.fastly.net, DNS:a.sni.global-ssl.fastly.net

Seems it's temporary issue from npm side and resolved now. I have checked it in my local and it's working as expected.
Please refer discussion -> https://npm.community/t/registry-npmjs-org-certification-invalid/1860/5

**Comment by krzysiekpiasecki** [comment]
@aagarwal2286  You were right. It works.

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "start": "webpack-dev-server",
     "dev": "webpack-dashboard -t 'Project' -- webpack-dev-server",
+    "docker-dev": "webpack-dev-server",
```
Comment (krzysiekpiasecki) [inline package.json]: Duplicate of start script

File: README.md [hunk 2]
```diff
@@ -39,6 +39,14 @@ Start by typing the following commands in your terminal in order to get **Admina
 > npm run dev
 ```
 
+#### Running with Docker
+```
```
Comment (krzysiekpiasecki) [inline README.md:43]: Add minimum compatibility version for docker binary

File: init.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (krzysiekpiasecki) [inline init.sh:1]: The script name is to general IMHO. This script modifies global git settings and can be executed out of the docker context. It's reasonable to name it e.g. docker-init.sh. I'm not to much familiar with the docker. Is it possible to inline this three commands into a configuration file?

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+services:
+  ui:
+    build:  
+      context: .
+      dockerfile: Dockerfile  
+    command: "sh init.sh"    
+    working_dir: /home/node/app
+    volumes:
+      - ${PWD}/:/home/node/app
+    container_name: ui
```
Comment (krzysiekpiasecki) [inline docker-compose.yaml:11]: s/ui/adminator/g

File: package.json (hunk 1 above)
Comment (krzysiekpiasecki) [inline package.json:9]: Duplicate of start script

**Review by krzysiekpiasecki — CHANGES_REQUESTED** [review]
Successfully tested on Mac OS environment. After merge it could be added to docker hub.

File: README.md (hunk 2 above)
Comment (krzysiekpiasecki) [reply README.md:43]: Files/Folders Structure needs update too.

**Review by krzysiekpiasecki — CHANGES_REQUESTED** [review]

**Comment by jepperaskdk** [comment]
Thanks for contributing this. I'm wondering if it is necessary though, as our only dependency is node-12.
You could just run the node-12 image directly, mounting your current directory (adminator) into /app:
`docker run -v $(pwd):/app -p 3000:3000 -it node:12-buster-slim /bin/bash`
Scripts for cloning the repository (init.sh) I wouldn't add in the repository.

Should we close as stale?

**Comment by jepperaskdk** [comment]
Closing, let me know if you think it should be reopened.

**Pull request closed without merging**